app/charts/map/map.tsx

```diff
@@ -63,21 +66,31 @@ export const MapComponent = () => {
     }
     mapNodeRef.current = mapRef;
   };
-
   const currentBBox = useRef<BBox>();
 
-  // Reset the map to its initial state.
-  const reset = () => {
-    if (initialViewState) {
-      const { longitude, latitude, zoom } = initialViewState;
+  const lockedRef = useRef(locked);
+  useEffect(() => {
+    lockedRef.current = locked;
+  }, [locked]);
+
+  // Resets the map to its default state (showing all visible features).
+  const reset = useCallback(() => {
+    // Reset the map only when it's in an unlocked mode.
+    if (!lockedRef.current) {
+      const { longitude, latitude, zoom } = defaultViewState;
       const newViewState = {
         center: [longitude, latitude] as LngLatLike,
         zoom,
-        duration: FLY_TO_DURATION,
+        duration: RESET_DURATION,
       };
       mapNodeRef.current?.flyTo(newViewState);
     }
-  };
+  }, [defaultViewState]);
```

Comment:
I think useEvent would be better to use here. Then you would not need the dependencies array, it would make it clearer what is the value of reset.

And then it would make the useEffect below also clearer since the dependencies of the useEffect could be `[defaultViewState]` and then it's clear that "when the default view state changes, we reset", otherwise we have to think "reset changes when default view state changes, so the use effect will be called when default view state changes".

Reply:
Yes, I also wasn't happy with this useEffect hook, thanks for the suggestion!